### ISSUE TITLE
Consider more redundancy in backup for better assurance

### ISSUE THREAD

**Issue by nobiot** [nobiot/org-transclusion#6]
Follow on from #5 as a backup solution.

Consider further https://www.emacswiki.org/emacs/backup-each-save.el

**Comment by nobiot** [comment]
Or https://www.emacswiki.org/emacs/ForceBackups

> By default, Emacs creates a backup only when you save the first time. This is incompatible with most people’s basic editing behavior of leaving Emacs and its buffers open all the time and constantly making changes and writing them to disk. Here we describe how to make Emacs do a backup on every save, not just the first.

**Comment by alphapapa** [comment]
I don't mean to tell you what to do, and I applaud your taking data integrity seriously, but I wonder if you should not try to force backups to be made.  Thinking as a user, I wouldn't want this package to dump extra copies of my files on the disk.  That could even be dangerous in the case of encrypted files, e.g. when using org-crypt to encrypt certain headings (maybe it would work as expected and save only the encrypted data, but it's complicated, and easy to make mistakes or misunderstand).

It seems to me that if you implement saving properly, it should work much like Org source blocks: when editing a block in a dedicated buffer, saving that buffer causes the changes to be propagated to the source buffer and the source buffer to be saved.  (To be fair, IIRC Org can save backup copies of the source blocks to files, and may even do so by default, but I discovered that by accident when I found copies of blocks in files that I didn't realize were being saved.)

To put it another way, I think a package like this should be as simple as possible, so it should use existing functionality as much as possible, and avoid adding extra functionality, like this, as much as possible.

Keep up the good work!

**Comment by nobiot** [comment]
@alphapapa Thank you for your comment and suggestions.

The potential security problem for encrypted files is something I never thought of.
I do not like the cluttering the file system, either. I am considering to make the "addtional backup" as an optional with default being switched off -- `org-src-edit` seems to do that too. Or completely remove it for now. 

I'm experimenting an alternative approach ([described here](https://org-roam.discourse.group/t/prototype-transclusion-block-reference-with-emacs-org-mode/830/42)):
![a00b853d72be6554d42db68d290e309cffb6743f](https://user-images.githubusercontent.com/12507865/97271602-10d83100-1831-11eb-8066-86cb68d1f380.jpeg)

I'm also looking at [John Kitchin's sample code](https://github.com/whacked/transclusion-minor-mode/blob/master/transcluding-org-elements.org) (via whacked), too.

With these, it seems that I can make the implementation a lot simpler and focused on what it should do. My local implementation is showing some good initial signs. I am hoping to show and tell the new version soon (with a short and crisp version of video next time 😉 ).

**Comment by alphapapa** [comment]
Do you mean you're considering abandoning the live-editing feature, so transcluded trees couldn't be edited in the receiving buffer?  That would certainly simplify the implementation, and I'd understand if limitations in Emacs made that necessary, but at the same time it would be a shame to lose that feature.  Anyway, keep up the good work.  :)

**Comment by nobiot** [comment]
Yes... Live, real-time sync is great looking, but perhaps not necessary. And it seems to introduce additional issues to resolve. I will see how the new approach will go. I think it is OK to explore different avenues -- I find that Emacs (Elisp) really enables us to do so rather efficiently.

**Comment by nobiot** [comment]
Implemented a new approach. I have put back a live-sync edit feature as an experiment. You can find more about the new approach and features in the new README.

Will do a video update soon. Thank you.
